tests/persistent_volumes_test.go

```diff
@@ -258,3 +266,196 @@ func TestPVCTemplateResize(t *testing.T) {
 	}
 
 }
+
+func TestPVCChangeStorage(t *testing.T) {
+	longOrSkip(t)
+
+	k8sNameSpace := getNamespace(t)
+	arangoClient := kubeArangoClient.MustNewInCluster()
+	kubecli := mustNewKubeClient(t)
+	mode := api.DeploymentModeCluster
+
+	defaultStorageClass := getDefaultStorageClassOrDie(t, kubecli)
+	randomString := strings.ToLower(uniuri.NewLen(4))
+	newStorageClassName := defaultStorageClass.GetName() + randomString
+
+	newStorage := defaultStorageClass.DeepCopy()
+	newStorage.ObjectMeta = metav1.ObjectMeta{
+		Name: newStorageClassName,
+	}
+	newStorage, err := kubecli.StorageV1().StorageClasses().Create(newStorage)
+	require.NoError(t, err)
+	defer func() {
+		err := kubecli.StorageV1().StorageClasses().Delete(newStorage.Name, &metav1.DeleteOptions{})
+		assert.NoError(t, err)
+	}()
+
+	name := strings.Replace(fmt.Sprintf("tcs-%s-%s", mode[:2], randomString), ".", "", -1)
+	depl, err := newDeploymentWithValidation(name, func(deployment *api.ArangoDeployment) {
+		var agentsCount, coordinatorCount, DBServersCount = 3, 2, 3
+
+		deployment.Spec.Mode = api.NewMode(mode)
+		deployment.Spec.Environment = api.NewEnvironment(api.EnvironmentProduction)
+
+		volumeMode := corev1.PersistentVolumeFilesystem
+		deployment.Spec.DBServers.VolumeClaimTemplate = &corev1.PersistentVolumeClaim{
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					corev1.ReadWriteOnce,
+				},
+				VolumeMode: &volumeMode,
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceStorage: resource.MustParse("1Gi"),
+					},
+				},
+			},
+		}
+
+		deployment.Spec.DBServers.Count = util.NewInt(DBServersCount)
+		deployment.Spec.Agents.Count = util.NewInt(agentsCount)
+		deployment.Spec.Coordinators.Count = util.NewInt(coordinatorCount)
+	})
+	require.NoError(t, err)
+
+	// Create deployment
+	_, err = arangoClient.DatabaseV1alpha().ArangoDeployments(k8sNameSpace).Create(depl)
+	require.NoError(t, err, "failed to create deployment: %s", err)
+	defer deferedCleanupDeployment(arangoClient, depl.GetName(), k8sNameSpace)
+
+	depl, err = waitUntilDeployment(arangoClient, depl.GetName(), k8sNameSpace, deploymentIsReady())
+	require.NoError(t, err, fmt.Sprintf("Deployment not running in time: %s", err)) // <-- fails here at the moment
+	require.NotNil(t, depl.Spec.DBServers.VolumeClaimTemplate)
+
+	// Update deployment
+	depl.Spec.DBServers.VolumeClaimTemplate.Spec.StorageClassName = util.NewString(newStorageClassName)
+	_, err = arangoClient.DatabaseV1alpha().ArangoDeployments(k8sNameSpace).Update(depl)
```

Comment:
This looks unstable, because the operator could have updated the spec and thus reject the update because of an old revision? Just asking ...

Reply:
changed